structuralcodes/sections/_generic.py

```diff
@@ -775,26 +775,92 @@ def calculate_moment_curvature(
         return res
 
     def calculate_nm_interaction_domain(
-        self, theta: float = 0, num_axial: int = 100
+        self,
+        theta: float = 0,
+        n1: int = 1,
+        n2: int = 2,
+        n3: int = 15,
+        n4: int = 10,
+        n5: int = 3,
+        n6: int = 4,
+        n: t.Optional[int] = None,
+        type1: t.Literal['linear', 'geometric', 'quadratic'] = 'linear',
+        type2: t.Literal['linear', 'geometric', 'quadratic'] = 'linear',
+        type3: t.Literal['linear', 'geometric', 'quadratic'] = 'geometric',
+        type4: t.Literal['linear', 'geometric', 'quadratic'] = 'linear',
+        type5: t.Literal['linear', 'geometric', 'quadratic'] = 'linear',
+        type6: t.Literal['linear', 'geometric', 'quadratic'] = 'linear',
```

Comment:
This is pretty clear. However, I would prefer calling the kwargs `num_1` and `type_1` etc. for consistency with other API.